github_rules/github_repo_initial_access.yml

```diff
@@ -0,0 +1,78 @@
+AnalysisType: rule
+Filename: github_repo_initial_access.py
+RuleID: GitHub.Repo.InitialAccess
+DisplayName: GitHub Branch Protection Policy Override
+Enabled: false
+LogTypes:
+  - GitHub.Audit
+Tags:
+  - GitHub
+Severity: Info
+Description: Detects when a user intially access a private organization repository.
+SummaryAttributes:
+  - p_any_ip_addresses
+Tests:
+  -
+    Name: GitHub - Initial Access
+    ExpectedResult: true
+    Mocks:
+      - objectName: get_string_set
+        returnValue: ""
+      - objectName: put_string_set
+        returnValue: ""
+    Log:
+      {
+        "@timestamp": 1623971719091,
+        "business": "",
+        "org": "my-org",
+        "repo": "my-org/my-repo",
+        "action": "git.push",
+        "p_log_type": "GitHub.Audit",
+        "protocol_name": "ssh",
+        "repository": "my-org/my-repo",
+        "repository_public": false,
+        "actor": "cat",
+        "user": ""
+      }
+  -
+    Name: GitHub - Repeated Access
+    ExpectedResult: false
+    Mocks:
+      - objectName: get_string_set
+        returnValue: "{\"cat\":\"my-repo\"}"
```

Comment:
*chef's kiss*

Although if you want to save yourself the trouble on larger mocked json strings YAML provides some options:

```
returnValue: '{"cat":"my-repo"}'
```

or I believe even this will work:

```
returnValue: >
  {"cat":"my-repo"}
```